### ISSUE TITLE
Disk space growing rapidly because of daily bulk import with ~40GB JSONL file and 25 million documents

### ISSUE THREAD

**Issue by kdevan** [typesense/typesense#2005]
## Description

The 40GB JSONL file is about 20 to 25 million rows/documents. It upserts just fine using the `parallel` command as recommended in the docs. The indexing takes quite some time but that's expected. This is a process that runs at least once a day, sometimes twice a day. The file contains many documents that are already part of the index and may just need to be updated so we use upsert.

The unexpected part I'm running into is that the result set fits into about 20GB of RAM but after a few days of running, the volume grows massively. After two or three days the volume is up to 255 GB. The volume is set to auto-expand but this is unsustainable cost wise. I see that the disk is supposed to be about 5x more than the RAM size so this seems out of the ordinary.

What could be causing the larger than normal disk growth? I assume that doing such big bulk inserts every day is not normal usage of Typesense so I'm guessing the issue has something to do with that. 

- Is there any recommended settings that might help with bulk inserts like this? 
- Is there any further debugging I can do to figure out why the disk is growing at such a rapid rate?

The analytics, data, and log are all set to the same volume which looks like this:

```
4.1M	analytics/
224G	lib/
38M     log/
```

Metrics just in case it helps (it's currently indexing):

```json
{
  "system_cpu1_active_percentage": "88.89",
  "system_cpu2_active_percentage": "100.00",
  "system_cpu3_active_percentage": "0.00",
  "system_cpu4_active_percentage": "0.00",
  "system_cpu_active_percentage": "46.15",
  "system_disk_total_bytes": "358194634752",
  "system_disk_used_bytes": "254468521984",
  "system_memory_total_bytes": "20979916800",
  "system_memory_total_swap_bytes": "2147479552",
  "system_memory_used_bytes": "14086385664",
  "system_memory_used_swap_bytes": "2146586624",
  "system_network_received_bytes": "51087",
  "system_network_sent_bytes": "162455",
  "typesense_memory_active_bytes": "15143034880",
  "typesense_memory_allocated_bytes": "13781464832",
  "typesense_memory_fragmentation_ratio": "0.09",
  "typesense_memory_mapped_bytes": "15500439552",
  "typesense_memory_metadata_bytes": "233886864",
  "typesense_memory_resident_bytes": "15143034880",
  "typesense_memory_retained_bytes": "3874099200"
}
```

## Steps to reproduce

I could provide the JSONL file if that helps.

## Expected Behavior

The disk should remain around 20% more than the RAM usage.

## Actual Behavior

The disk is about 12x the RAM usage.

## Metadata

**Typesense Version**: 27.1

**OS**: Debian bookworm

**Comment by jasonbosco** [comment]
Could you try increasing the disk compacting frequency using the `db-compaction-interval` parameter documented under [this table](https://typesense.org/docs/27.1/api/server-configuration.html#resource-usage) and/or run [manual compaction](https://typesense.org/docs/27.1/api/cluster-operations.html#compacting-the-on-disk-database)?

**Comment by kishorenc** [comment]
During indexing, there are a couple of things that are happening which will increase disk size:

1. The incoming writes are buffered in the raft queue (which is shipped to other nodes of the cluster for replication)
2. We buffer the incoming import/write data in the rocksdb database.

The raft queue is purged hourly  when the snapshot runs, as controlled by the `--snapshot-interval-seconds` flag. 

The buffered writes are deleted once the import is done, but this is logical deletion, and the actual purge from disk happens during rocksdb compaction (as controlled by the `db-compaction-interval` flag or via the API). We also keep a backup of last 2 snapshots, so even after compaction, it will take an hour or two for the disk size change to be fully realized. 

To summarize, if you just call the compaction API call and waited for a couple of hours (for the older snapshot backups to expire), the disk size will return to baseline size.
